cpp/src/interop/to_arrow.cu

```diff
@@ -176,6 +176,40 @@ std::shared_ptr<arrow::Array> dispatch_to_arrow::operator()<numeric::decimal64>(
   return std::make_shared<arrow::Decimal128Array>(data);
 }
 
+template <>
+std::shared_ptr<arrow::Array> dispatch_to_arrow::operator()<numeric::decimal128>(
+  column_view input,
+  cudf::type_id,
+  column_metadata const&,
+  arrow::MemoryPool* ar_mr,
+  rmm::cuda_stream_view stream)
+{
+  using DeviceType = __int128_t;
+
+  rmm::device_uvector<DeviceType> buf(input.size(), stream);
+
+  thrust::copy(rmm::exec_policy(stream),  //
+               input.begin<DeviceType>(),
+               input.end<DeviceType>(),
+               buf.begin());
+
+  auto const buf_size_in_bytes = buf.size() * sizeof(DeviceType);
+  auto data_buffer             = allocate_arrow_buffer(buf_size_in_bytes, ar_mr);
+
+  CUDA_TRY(cudaMemcpyAsync(data_buffer->mutable_data(),
```

Comment:
Why do we stage the data to a `device_uvector` before the copy? Why not copy directly from the column?

Reply:
I am not exactly sure what you mean. Can you elaborate?

Reply:
If I am reading this correctly, we manifest a `device_uvector` and copy the column into that and then we turn around and memcpy the `device_uvector` out to an arrow pointer. I am asking why we can't do the `cudaMemcpyAsync` call with a pointer to the cudf column itself instead of the `device_uvector`. Am I missing a transform of some sort?